contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -0,0 +1,381 @@
+         * is correctly set
+         */
+        _approveBase();
+    }
+
+    /**
+     * @dev Deposit asset into the Curve ETH pool
+     * @param _weth Address of WETH
+     * @param _amount Amount of asset to deposit
+     */
+    function deposit(address _weth, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_weth, _amount);
+    }
+
+    function _deposit(address _weth, uint256 _wethAmount) internal {
+        require(_wethAmount > 0, "Must deposit something");
+        require(_weth == address(poolWETHToken), "Can only deposit WETH");
+
+        emit Deposit(_weth, address(lpToken), _wethAmount);
+
+        // safe to cast since min value is at least 0
+        uint256 oethToAdd = uint256(
+            _max(
+                0,
+                int256(curvePool.balances(wethCoinIndex)) +
+                    int256(_wethAmount) -
+                    int256(curvePool.balances(oethCoinIndex))
+            )
+        );
+
+        /* Add so much OETH so that the pool ends up being balanced. And at minimum
+         * add as much OETH as WETH and at maximum twice as much OETH.
+         */
+        oethToAdd = Math.max(oethToAdd, _wethAmount);
+        oethToAdd = Math.min(oethToAdd, _wethAmount * 2);
+
+        /* Mint OETH with a strategy that attempts to contribute to stability of OETH/WETH pool. Try
+         * to mint so much OETH that after deployment of liquidity pool ends up being balanced.
+         *
+         * To manage unpredictability minimal OETH minted will always be at least equal or greater
+         * to WETH amount deployed. And never larger than twice the WETH amount deployed even if
+         * it would have a further beneficial effect on pool stability.
+         */
+        if (oethToAdd > 0) {
+            IVault(vaultAddress).mintForStrategy(oethToAdd);
+        }
+
+        uint256[2] memory _amounts;
+        _amounts[wethCoinIndex] = _wethAmount;
+        _amounts[oethCoinIndex] = oethToAdd;
+
+        uint256 valueInLpTokens = (_wethAmount + oethToAdd).divPrecisely(
+            curvePool.lp_price()
+        );
+        uint256 minMintAmount = valueInLpTokens.mulTruncate(
+            uint256(1e18) - MAX_SLIPPAGE
+        );
+
+        uint256 balance = poolOETHToken.balanceOf(address(this));
+        // Do the deposit to Curve ETH pool
+        uint256 lpDeposited = curvePool.add_liquidity(_amounts, minMintAmount);
+        //uint256 lpDeposited = curvePool.add_liquidity(_amounts, uint256(0));
+        _lpDeposit(lpDeposited);
+    }
+
+    /**
+     * @dev Deposit the entire balance of any supported asset into the Curve 3pool
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 balance = poolWETHToken.balanceOf(address(this));
+        if (balance > 0) {
+            _deposit(address(poolWETHToken), balance);
+        }
+    }
+
+    function _lpDeposit(uint256 lpToDeposit) internal {
```

Comment:
I don't think we need _lpDeposit and _lpWithdrawAll as separate methods, since they are each only used once and are very simple.

Reply:
yup agreed have removed them